### ISSUE TITLE
Missing definition in API documentation: `repo.git`

### ISSUE THREAD

**Issue by cowlinator** [gitpython-developers/GitPython#1405]
API documentation at https://gitpython.readthedocs.io/en/stable/reference.html#git.repo.base.Repo.git does not describe `repo.git`.  What type of object is it? What are its function definitions?

I'm looking at modifying code that uses `repo.git.describe()` (which I would think would be a popular function), but I can find no function definition in the api, and I dont' know what arguments it accepts.

**Comment by Byron** [comment]
Indeed, the field itself could use some documentation maybe along with some docs on [how to use it](https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly).

It can't be documented as its functions are all 'virtual', they act as away to build a `git` command-line invocation, so `repo.git.foo()` is valid and causes `git foo` to be called.

**Comment by cowlinator** [comment]
@Byron , ok good to know.

And yes, that is exactly the information that needs to be in that documentation. 

```
repo.git is a virtual object which represents a git command-line invocation.  See https://gitpython.readthedocs.io/en/stable/tutorial.html#using-git-directly
```
